storage/watchable_store.go

```diff
@@ -441,3 +454,35 @@ func unsafeAddWatching(synced *map[string]map[*watching]struct{}, k string, wa *
 	mp[k][wa] = struct{}{}
 	return nil
 }
+
+// mapWatchingToEvents maps watchings to events.
+func mapWatchingToEvents(sm map[string]map[*watching]struct{}, evs []storagepb.Event) map[*watching][]storagepb.Event {
```

Comment:
newWatchingToEventMap creates a map that has watching as key and events as value. It enables quick events look up by watching.